core/src/main/java/com/datastax/oss/driver/api/core/config/DriverConfigLoader.java

```diff
@@ -58,13 +62,27 @@
    */
   @NonNull
   static DriverConfigLoader fromClasspath(@NonNull String resourceBaseName) {
+    return fromClasspath(resourceBaseName, DRIVER_CLASS_LOADER);
+  }
+
+  /**
+   * Just like {@link #fromClasspath(java.lang.String)} except that application configuration
+   * resources will be located using the provided {@link ClassLoader} instead of the driver's
+   * {@linkplain DriverConfigLoader#DRIVER_CLASS_LOADER default one}.
+   */
+  @NonNull
+  static DriverConfigLoader fromClasspath(
+      @NonNull String resourceBaseName, @NonNull ClassLoader appClassLoader) {
     return new DefaultDriverConfigLoader(
         () -> {
           ConfigFactory.invalidateCaches();
```

Comment:
This is unrelated but I think we should improve cache invalidation, it's being done more than once in certain cases. I think it's the config loader's responsibility to invalidate the caches, not the config suppliers'.

Reply:
Do you want me to create a new ticket for this, or address it in this PR?

Reply:
New ticket, because this PR should be released in 4.5.1. BTW, I think you need to change the base branch to 4.5.x, right @olim7t ?

Reply:
Will do to both.

Reply:
JAVA-2689 created for the cache improvements.

Reply:
Thank you!

Reply:
This has shifted to 4.6.0 now, don't forget to change the base branch again.

Reply:
:+1:  I'll rebase this branch with the next commits I push, and then change the PR target branch back to 4.x